src/context/LoginContext/loginReducer.ts

```diff
@@ -0,0 +1,57 @@
+import { loginInitialState, User } from './LoginContext';
+
+type ActionMap<M extends { [index: string]: any }> = {
+  [Key in keyof M]: M[Key] extends undefined
+    ? {
+        type: Key;
+      }
+    : {
+        type: Key;
+        payload?: M[Key];
+      };
+};
+
+export enum Types {
+  SET_USER = 'SET_USER',
+  DELETE_USER = 'DELETE_USER',

```

Comment:
Reducer action types should be event based. If you name it this way it kind of ties you to what the action can do. It can set the user. If you have a type like this and you don't _just_ set or delete a user then the name is misleading. Instead if you named this `LOGGED_IN` or `LOGGED_OUT` you can adjust the state accordingly based on the action/event that dispatched

Reply:
understood, and yes you are right makes sense, my bad here, i will use LOGIN_USER and LOGOUT_USER actions